styx-scheduler-service/src/main/java/com/spotify/styx/state/QueuedStateManager.java

```diff
@@ -288,8 +298,12 @@ private void process(RunState state, long counterPosition, Event event,
       throw t;
     }
 
+    // If the service is stopped right before this line, the event was successfully stored but never
+    // going to be consumed. Event consumption is only best effort with the current implementation.
+    consumeEvent(sequenceEvent);
```

Comment:
I would inline this function unless it's used elsewhere. It looks weird to have only this one in a function and the other one inlined (the `outputHandlerExecutor.execute` one).